### ISSUE TITLE
Improve "Devices & apps" section - attach apps/services to devices

### ISSUE THREAD

**Issue by LZoog** [mozilla/fxa#3577]
This issue may need to be split into two or more smaller issues when more details are laid out. @ryanfeeley please feel free to edit this as you see fit.

[Per Ryan in #3543](https://github.com/mozilla/fxa/pull/3543#issuecomment-561253432_) (PR), #2505 (issue) was intended as a half-measure to get us to the following design, in which we show all services attached to user devices:

<img width="609" alt="image" src="https://user-images.githubusercontent.com/68213/70070674-fe725a80-15c1-11ea-8f4a-0f6dfc8c6095.png">

However, that issue and associated PR were determined to not have the desired half-measure effect and instead raised several questions.
* What apps, if any (besides a browser session), actually make sense to display "Web Session" text next to their names? Does this give users insight or benefit?
* What apps are we _not_ collecting a `deviceId` or displaying a device name when we should be? At a minimum, we are not collecting `deviceId` for Firefox Private Network, and [Firefox Lockwise doesn't set a custom device name](https://github.com/mozilla/fxa/pull/3543#issuecomment-560557497).

If we have a complete list of devices & apps that can be displayed here and hand that over to UX (or does UX already own a list?) to create a mockup of how each is desired to be displayed and compare to how we display them now, we can begin to break apart the work needed here. UX can also consider the settings redesign approaching in Q1.

And, lower-level comments/questions:
* This seems inconsistent: `Web Session, Firefox 70` vs `Firefox on Fox’s iPad`
* Should `scope` be displayed in `title` attributes? What purpose does this serve?
<img width="532" alt="image" src="https://user-images.githubusercontent.com/13018240/70066847-ff4fc000-15b2-11ea-8cd8-d4d0263f69f2.png">

---

To summarize, at the time of issue creation, it seems to me that we mostly need to determine the scope of work around attaching apps/services to associated devices, pending the final design. Per @rfk [in this comment](https://github.com/mozilla/fxa/pull/3543#issuecomment-560962333), echoing @shane-tomlinson's [detailed comment](https://github.com/mozilla/fxa/issues/2505#issuecomment-531698737), we will need to at least [associate sessionTokens with access/refreshTokens](https://github.com/mozilla/fxa/issues/2479), which at the time of writing, has an estimate of 21.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-52)
┆Issue Number: FXA-52

**Comment by johngruen** [comment]
@davismtl when we create a settings epic, probably add this.

**Comment by LZoog** [comment]
Note: "Web Session, Safari 13" is displayed when users login with an iPad on Safari, but the device appears to be captured when logging in on Firefox (maybe this is intentional?). One layer of complication around this issue is that as of iOS 13, Safari on an iPad and Safari on a Mac send exactly the same user agent header, upon which our attached clients depend on:

```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.4 Safari/605.1.15
```

iPad login vs desktop login:
![image](https://user-images.githubusercontent.com/13018240/70820118-7ba98680-1d9d-11ea-8cd0-08660d8046ed.png)


This appears to be intentional by Apple to force a more desktop-like experience for iPads instead of a tablet-specific experience.
The only solution for this I've seen suggested upon searching is a client-side check for `navigator.maxTouchPoints > 0` to check if the device is a touch screen. I don't really like the idea of checking for this and sending/storing that data because it would become useless as soon as Apple comes out with a touch screen desktop device. Thankfully, the Firefox user agent, while sending a desktop OS, includes `FxiOS` that we can use to differentiate. This iPad Safari-specific issue might not be worth addressing.

**Comment by LZoog** [comment]
Another possible inconsistency: I see location on at least one of these but not others.

![image](https://user-images.githubusercontent.com/13018240/71364347-8d3f1b00-2561-11ea-957e-971f5745c85c.png)

**Comment by janiceahn** [comment]
I believe that Ana Mediniac (sp?) is working on a related UR report that may inform this bug

**Comment by AnaMedinac** [comment]
The UR report I'm working on focuses on the users mental model around Logging out.  Currently the UR looks into the "Devices & Apps" section, however, it only explores the users high level mental models and understanding of the `Disconnect` and `Revoke` buttons. 

We don't go into any detail around apps and services and attaching them to associated devices, although that could make for a cool study in the future to determine the benefit it provides to users.

**Comment by data-sync-user** [comment]
➤ Vijay Budhram commented:

Closing this is an old settings bug.
